src/AElf.Kernel.CodeCheck/Application/CodeCheckValidationProvider.cs

```diff
@@ -12,15 +12,17 @@ internal class CodeCheckValidationProvider : IBlockValidationProvider
     private readonly ICheckedCodeHashProvider _checkedCodeHashProvider;
     private readonly IContractReaderFactory<ACS0Container.ACS0Stub> _contractReaderFactory;
     private readonly ISmartContractAddressService _smartContractAddressService;
+    private readonly ICodeCheckService _codeCheckService;
 
     public CodeCheckValidationProvider(ISmartContractAddressService smartContractAddressService,
```

Comment:
**A question (for thought only, may not be resolved in this PR):**

What's the relationship between `Provider` and `Service`? Does it make sense to have some `Provider`s calling some `Service`s while some other `Service` are calling some `Provider`s?

Reply:
I'm not sure about that, but I think `Provider` can only be called from its own module.